app/application/use_cases/add_vault_member.py

```
@@ -0,0 +1,143 @@
+
+        # Step 2: Verify actor is the owner (only owners can add members)
+        if vault.owner_id != inp.actor_user_id:
+            raise UnauthorizedVaultAccessError(inp.actor_user_id, inp.vault_id)
+
+        # Step 3: Prevent adding owner as a member (they already have access)
+        # This is a business rule to maintain clear ownership semantics
+        if inp.target_user_id == vault.owner_id:
+            raise ValueError(
+                "Owner already has implicit access and cannot be added as a member. "
+                "Use the vault's owner_id for ownership operations."
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)